src/main/java/minimalisp/Lisp.java

```diff
@@ -341,6 +341,15 @@ public int compare(T o1, T o2) {
   }
 
   /**
+   * Returns a new Map with the contents of the second provided map merged over the provided map  
+   */
+  public static <K, V> Map<K, V> merge(Map<K, V> a, Map<K, V> b) {
+    Map<K, V> merged = new LinkedHashMap<K, V>(a);
```

Comment:
What if you wanted a different kind of map as the result, e.g. a SortedMap of some kind? I use TreeMap a lot.

Could we have an overloaded variant that passes in the target map as a third argument? You could use like this: merge(a, b, treemap())

If you called merge(a, b) it would call merge(a, b, map()), or maybe merge(a, b, linkedhashmap())

If you agree, we could potentially apply this approach to other methods that create new collections.